ltr/sagemaker/deploy.py

```diff
@@ -0,0 +1,115 @@
+import sagemaker
+import sagemaker.tensorflow
+import sys
+import time
+
+def deploy(model_name, s3_bucket, role, govuk_environment, instance_count, instance_type):
+    endpoint_name = f"govuk-{govuk_environment}-search-ltr-endpoint"
+
+    session = sagemaker.Session()
+
+    update_endpoint = False
+
+    # try to find the current endpoint config
+    try:
+        current_endpoint_config_name = session.boto_session.client(
+            "sagemaker"
+        ).describe_endpoint(EndpointName=endpoint_name)["EndpointConfigName"]
+
+        current_endpoint_config = session.boto_session.client(
+            "sagemaker"
+        ).describe_endpoint_config(EndpointConfigName=current_endpoint_config_name,)
+
+        update_endpoint = True
+    except Exception:
```

Comment:
This catches absolutely everything, including potentially things like syntax errors, calls to nonexistent methods if there are typos, etc. You probably want to catch some slightly more specific type (probably `botocore.exceptions.ClientError`?)

Reply:
Sorry should have made it clearer - this code is originally from the "ltr/jenkins/deploy.py" file. I've just wrapped it in the deploy function to so we can call it from main.py. I'll try and tidying this code up in a separate PR to address these comments.

Reply:
Ohh sorry, no probs at all!